src/Microsoft.AspNetCore.Localization/ApplicationBuilderExtensions.cs

```diff
@@ -43,12 +43,77 @@ public static IApplicationBuilder UseRequestLocalization(this IApplicationBuilde
             {
                 throw new ArgumentNullException(nameof(app));
             }
+            
             if (options == null)
             {
                 throw new ArgumentNullException(nameof(options));
             }
 
             return app.UseMiddleware<RequestLocalizationMiddleware>(Options.Create(options));
         }
+
+        /// <summary>
+        /// Adds the <see cref="RequestLocalizationMiddleware"/> to automatically set culture information for
+        /// requests based on information provided by the client. 
+        /// </summary>
+        /// <param name="app">The <see cref="IApplicationBuilder"/>.</param>
+        /// <param name="optionsAction"></param>
+        /// <returns>The <see cref="IApplicationBuilder"/>.</returns>
+        public static IApplicationBuilder UseRequestLocalization(
+            this IApplicationBuilder app,
+            Action<RequestLocalizationOptions> optionsAction)
+        {
+            if (app == null)
+            {
+                throw new ArgumentNullException(nameof(app));
+            }
+
+            if (optionsAction == null)
+            {
+                throw new ArgumentNullException(nameof(optionsAction));
+            }
+
+            var options = new RequestLocalizationOptions();
+            optionsAction.Invoke(options);
+
+            return app.UseMiddleware<RequestLocalizationMiddleware>(Options.Create(options));
+        }
+
+        /// <summary>
+        /// Adds the <see cref="RequestLocalizationMiddleware"/> to automatically set culture information for
+        /// requests based on information provided by the client. 
+        /// </summary>
+        /// <param name="app">The <see cref="IApplicationBuilder"/>.</param>
+        /// <param name="cultures">The culture names to be added by the application, which is represents both supported cultures and UI cultures.</param>
+        /// <returns>The <see cref="IApplicationBuilder"/>.</returns>
+        /// <remarks>
+        /// Note that the first culture is the default culture name.
+        /// </remarks>
+        public static IApplicationBuilder UseRequestLocalization(
+            this IApplicationBuilder app,
+            params string[] cultures)
+        {
+            if (app == null)
+            {
+                throw new ArgumentNullException(nameof(app));
+            }
+
+            if (cultures == null)
+            {
+                throw new ArgumentNullException(nameof(cultures));
+            }
+
+            if (cultures.Length == 0)
+            {
+                throw new ArgumentException(nameof(cultures));
+            }
+
+            var options = new RequestLocalizationOptions()
+                .AddSupportedCultures(cultures)
+                .AddSupportedUICultures(cultures)
+                .SetDefaultCulture(cultures[0]);
```

Comment:
I'm not in love with the idea of taking the first culture as the default. It seems undiscoverable/confusing. I would suggest just having a defaultCulture parameter even if it does lead to a little repetition. @rynowak what do you think?

Reply:
It comes down to scenarios IMO - do you **always** want a default culture?

Reply:
As it is now if you don't have a default culture set and a culture can't be determined from the request then it seems the culture the user receives will depend on the default THREAD culture, which I feel like is a scary and confusing fallback story. See [here](https://github.com/aspnet/Localization/blob/8b03ee8c81b61a852a586740c4db7b09c17f5e76/src/Microsoft.AspNetCore.Localization/RequestLocalizationMiddleware.cs#L146).

Reply:
Which is a good reason to have a sensible default. What would the API look like with an additional parameter? 

I'd actually suggest baking that behavior into the middleware rather than the options/builder. it seems like **anyone** would agree that's better than the current behavior. Is there a historical reason for this?

Reply:
I don't know the history behind why things are the way the currently are. @DamianEdwards might. I can get on-board with with putting this in the middle-ware.